.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+on: [pull_request, push]
+name: Tests
+
+jobs:
+  build:
+    name: Run tests
+    runs-on: ubuntu-latest
```

Comment:
should be ran on all 3 os-es. We can reduce build time to only run on all 3 os-es on PR and only ubuntu on push but since it's oss, it's free anyways